ga4gh/datamodel/variants.py

```diff
@@ -342,3 +343,118 @@ def getMetadata(self):
+        return [str(v) for v in value]
+    else:
+        return [str(value)]
+
+
+_nothing = object()
+
+
+def is_empty_iter(it):
+    """Return True iff the iterator is empty or exhausted"""
+    return next(it, _nothing) is _nothing
+
+
+class HtslibVariantSet(VariantSet):
+    """
+    Class representing a single variant set backed by a directory of indexed
+    VCF or BCF files.
+    """
+
+    def __init__(self, variantSetId, vcfPath, use_bcf=True):
+        self._variantSetId = variantSetId
+        self._created = protocol.convertDatetime(datetime.datetime.now())
+        self._chromFileMap = {}
+
+        for pattern in ['*.bcf', '*.vcf.gz']:
+            for filename in glob.glob(os.path.join(vcfPath, pattern)):
+                self._addFile(filename)
+
+    def _addFile(self, filename):
+        varFile = pysam.VariantFile(filename)
+        if not varFile.index:
+            raise Exception("VCF/BCF files must be indexed")
+        for chrom in varFile.index:
+            # Unlike Tabix indices, CSI indices include all contigs defined
+            # in the BCF header.  Thus we must test each one to see if
+            # records exist or else they are likely to trigger spurious
+            # overlapping errors.
+            if is_empty_iter(varFile.fetch(chrom)):
+                continue
+            if chrom in self._chromFileMap:
+                raise Exception("cannot have overlapping VCF/BCF files.")
+            self._chromFileMap[chrom] = varFile
+
+    def convertVariant(self, record):
+        variant = protocol.GAVariant()
+
+        # N.B. record.pos is 1-based
+        variant.id = "{0}:{1}:{2}".format(self._variantSetId,
+                                          record.contig,
+                                          record.pos)
+
+        # TODO How should we populate these from VCF?
+        variant.created = self._created
+        variant.updated = self._created
+
+        variant.variantSetId = self._variantSetId
+        variant.referenceName = record.contig
+        variant.names = record.id.split(';') if record.id else None
+        variant.start = record.start          # 0-based inclusive
+        variant.end = record.stop             # 0-based exclusive
+        variant.referenceBases = record.ref
+        variant.alternateBases = list(record.alts)
+
+        # record.filter and record.qual are also available, when supported
+        # by GAVariant.
+
+        variant.info = {}
+        for key, value in record.info.iteritems():
+            if value is not None:
+                variant.info[key] = _encode_value(value)
+
+        variant.calls = []
+        for name, call in record.samples.iteritems():
+            c = protocol.GACall()
+            c.genotype = _encode_value(call.allele_indices)
+
+            for key, value in record.info.iteritems():
+                if key == 'GT':
+                    c.genotype = list(call.allele_indices)
+                elif key == 'GL' and value:
```

Comment:
Can you rewrite this without the implicit boolean conversion please? (i.e., `and value is not None`)